src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactor/BackupSecondFactorActions.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button, BorderBoxProps, Flex } from "@artsy/palette"
+import React, { useEffect, useState } from "react"
+
+interface BackupSecondFactorActionsProps extends BorderBoxProps {
+  backupSecondFactors: string[]
+}
+
+export const BackupSecondFactorActions: React.FC<BackupSecondFactorActionsProps> = props => {
+  const { backupSecondFactors } = props
+
+  const [supportsClipboard, setSupportsClipboard] = useState(false)
+
+  useEffect(() => {
+    // Only render the copy button if browser supports the Clipboard API
+    // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
+    if ("clipboard" in navigator) setSupportsClipboard(true)
+  }, [])
+
+  function copyCodesToClipboard() {
+    navigator.clipboard.writeText(backupSecondFactors.join("\n"))
+  }
+
+  function downloadCodes() {
```

Comment:
Is there a security risk we need to consider in using a data blog for sensitive values? One thing is I believe you need to explicitly revoke the URL object to prevent memory leaks or unintended persistence. Any other security issues?

Good reference here: https://github.com/artsy/force/blob/master/src/v2/Components/UserSettings/TwoFactorAuthentication/Components/SmsSecondFactor/Modal.tsx#L77

Reply:
Hey @mc-jones , I guess the wrong link was pasted here by accident. This is the one contained in the other comment, related to the mutation handling.

I pushed this, though: 047ff31379846a01a446c899ea3473230e8f127d
Is this what you meant?

Reply:
Hey @lidimayra. Sorry about the wrong link! Here is the correct one for reference: https://www.w3.org/TR/FileAPI/#creating-revoking. 

But as discussed, I'll take finish out from here once the migration from @The-Beez-Kneez  is done.